src/EditProvider.ts

```diff
@@ -99,16 +100,84 @@ export class EditProvider
   }
 
   public static beautifyOptions(
-    filePath: string = vscode.workspace.rootPath
+    path: string = vscode.workspace.rootPath
   ): Promise<LanguageOptionValues> {
     try {
-      const explorer = cosmiconfig("unibeautify", {});
+      /**
+       * Use a different `.unibeautify` config when vscode config is set and
+       * define if the project config should extend the vscode config.
+       *
+       * @argument {unibeautify.defaultConfig} string path to a file
+       * @argument {unibeautify.extendConfig} boolean extend or not
+       */
+      const vscodeSettings = vscode.workspace.getConfiguration("unibeautify");
+      let defaultConfigOverride = vscodeSettings.defaultConfig;
+      let extendConfig = vscodeSettings.extendConfig;
+      let workspace = path;
+
+      const explorerOptions: any = {};
+      if (defaultConfigOverride) {
+        path = defaultConfigOverride;
+      }
+
+      const explorer = cosmiconfig("unibeautify", explorerOptions);
       const defaultConfig: LanguageOptionValues = {};
-      return explorer
-        .search(filePath)
-        .then(result => (result ? result.config : defaultConfig));
+
+      let isDirectory = fs.lstatSync(path).isDirectory();
+      if (!isDirectory) {
+        return explorer
+          .load(path)
+          .then(returnResult)
+          .catch(catchError);
+      } else {
+        return explorer
+          .search(path)
+          .then(returnResult)
+          .catch(catchError);
+      }
+
+      function returnResult(result: any) {
```

Comment:
The `beautifyOptions` function is growing too large. It should either be broken down into multiple methods or perhaps some of the code can be deleted and/or simplified. Such as by removing support for `unibeautify.defaultConfig` to be a directory.

Reply:
I tried to shorten it to this:

```
      return explorer
        .load(pathToFileOrDirectory)
        .then(result => (result ? result.config : defaultConfig)
        .catch(() => vscode.window.showErrorMessage(`We could not find your configuration file: ${path}. Please correct your path, otherwise the plugin will not work!`));
```

but I‘m getting this error:

```
[ts]
Der Typ "Promise<string | LanguageOptionValues>" kann dem Typ "Promise<LanguageOptionValues>" nicht zugewiesen werden.
  Der Typ "string | LanguageOptionValues" kann dem Typ "LanguageOptionValues" nicht zugewiesen werden.
    Der Typ "string" kann dem Typ "LanguageOptionValues" nicht zugewiesen werden.
```

Reply:
I do not speak that language so I cannot help you with analyzing the error message 😉 .
Take a look at https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#localized-diagnostics-on-the-command-line 😄

Reply:
@Glavin001 oh 😄 sorry, anyway I fixed it. btw it is german